core/domain/feedback_services_test.py

```diff
@@ -64,4 +65,188 @@ def test_status_of_newly_created_thread_is_open(self):
+        super(SuggestionServicesUnitTests, self).setUp()
+        with self.swap(feedback_models.FeedbackThreadModel, 
+                       'generate_new_thread_id', self._generate_thread_id):
+            (feedback_services.create_suggestion(
+                self.EXP_ID1, 'author_id', 3, 'state_name', {'old_content': {}}))
+        with self.swap(feedback_models.FeedbackThreadModel, 
+                       'generate_new_thread_id', self._generate_thread_id):
+            (feedback_services.create_suggestion(
+                self.EXP_ID2, 'author_id', 3, 'state_name', {'old_content': {}}))        
+        # Open thread with suggestion.
+        thread1 = feedback_models.FeedbackThreadModel(
+            id='.'.join([self.EXP_ID1, self.THREAD_ID1]),
+            exploration_id=self.EXP_ID1,
+            state_name='state_name',
+            original_author_id='author_id',
+            subject='SUGGESTION',
+            has_suggestion=True) 
+        # Closed threads with suggestion.
+        thread2 = feedback_models.FeedbackThreadModel(
+            id='.'.join([self.EXP_ID1, self.THREAD_ID2]),
+            exploration_id=self.EXP_ID1,
+            state_name='state_name',
+            original_author_id='author_id',
+            subject='SUGGESTION',
+            status=feedback_models.STATUS_CHOICES_FIXED,
+            has_suggestion=True)
+        thread3 = feedback_models.FeedbackThreadModel(
+            id='.'.join([self.EXP_ID1, self.THREAD_ID3]),
+            exploration_id=self.EXP_ID1,
+            state_name='state_name',
+            original_author_id='author_id',
+            subject='SUGGESTION',
+            status=feedback_models.STATUS_CHOICES_IGNORED,
+            has_suggestion=True)
+        # Closed thread without suggestion.
+        thread4 = feedback_models.FeedbackThreadModel(
+            id='.'.join([self.EXP_ID1, self.THREAD_ID4]),
+            exploration_id=self.EXP_ID1,
+            state_name='state_name',
+            original_author_id='author_id',
+            subject='SUGGESTION',
+            status=feedback_models.STATUS_CHOICES_IGNORED)
+        # Open thread without suggestion.
+        thread5 = feedback_models.FeedbackThreadModel(
+            id='.'.join([self.EXP_ID1, self.THREAD_ID5]),
+            exploration_id=self.EXP_ID1,
+            state_name='state_name',
+            original_author_id='author_id',
+            subject='SUGGESTION',
+            status=feedback_models.STATUS_CHOICES_OPEN)
+        self.THREADS = [thread1, thread2, thread3, thread4, thread5]
+
+    def test_create_and_get_suggestion(self):
+        with self.swap(feedback_models.FeedbackThreadModel, 
+                       'generate_new_thread_id', self._generate_thread_id):
+            (feedback_services.
+                create_suggestion(self.EXP_ID3, 'author_id', 3,
+                                  'state_name', {'old_content': {}}))
+            suggestion = feedback_services.get_suggestion(
```

Comment:
This and the following statements can move back outside the 'with' context, right?